modules/nlu-testing/src/views/full/TestModal.tsx

```diff
@@ -111,25 +70,13 @@ export const TestModal: FC<Props> = props => {
 
   const expectedIntentChanged = e => {
     const intent = intents.find(i => i.name === e.target.value)
-    setExpectedIntent(intent)
-  }
-
-  const testingCtxChanged = e => {
-    setTestingCtx(e.target.value)
-    setExpectedIntent(NONE_INTENT)
-  }
-
-  const slotConditionChanged = (slotName, e) => {
-    setSlotConditions({
-      ...slotConditions,
-      [slotName]: e.target.value
-    })
+    dispatch({ type: 'setExpectedIntent', data: { expectedIntent: intent } })
   }
 
   return (
     <Dialog
-      title={isEditing ? 'Edit test' : 'Create new test'}
-      icon={isEditing ? 'edit' : 'add'}
+      title={isEditing() ? 'Edit test' : 'Create new test'}
```

Comment:
Could you work isEditing back in the state? it seem weird to have a method call, maybe have a variable for title and icon?